src/routes/safe/components/Settings/SpendingLimit/NewSpendingLimit.tsx

```diff
@@ -0,0 +1,115 @@
+import { Button, Icon, Text, Title } from '@gnosis.pm/safe-react-components'
+import { Mutator } from 'final-form'
+import React from 'react'
+import GnoForm from 'src/components/forms/GnoForm'
+import GnoButton from 'src/components/layout/Button'
+import { Amount } from 'src/routes/safe/components/Settings/SpendingLimit/Amount'
+import BeneficiarySelect from 'src/routes/safe/components/Settings/SpendingLimit/BeneficiarySelect'
+import {
+  TitleSection,
+  StyledButton,
+  FooterSection,
+  FooterWrapper,
+} from 'src/routes/safe/components/Settings/SpendingLimit/index'
+import ResetTime from 'src/routes/safe/components/Settings/SpendingLimit/ResetTime'
+import TokenSelect from 'src/routes/safe/components/Settings/SpendingLimit/TokenSelect'
+import styled from 'styled-components'
+
+const FormContainer = styled.div`
+  padding: 24px;
+  align-items: center;
+  display: grid;
+  grid-template-columns: 4fr 1fr;
+  grid-template-rows: 6fr;
+  gap: 16px 8px;
+  grid-template-areas:
+    'beneficiaryInput beneficiaryScan'
+    'tokenInput .'
+    'amountInput .'
+    'resetTimeLabel resetTimeLabel'
+    'resetTimeToggle resetTimeToggle'
+    'resetTimeOption resetTimeOption';
+`
+
+const YetAnotherButton = styled(GnoButton)`
+  &.Mui-disabled {
+    background-color: ${({ theme }) => theme.colors.primary};
+    color: ${({ theme }) => theme.colors.white};
+    opacity: 0.5;
+  }
+`
+
+// TODO: propose refactor in safe-react-components based on this requirements
```

Comment:
not sure to understand

Reply:
mh! after the refactor that TODO lost its target. :smile: